### PR TITLE
Azure Software Stack and OS Image creation  and listing implemented

### PR DESCRIPTION
**Pull request by srikanthv02**
@udeshrl : Please review.

### CODE REVIEW

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 1]
```diff
@@ -4,10 +4,12 @@
         .controller('blueprintCreateCtrl',['$scope','$modal','toastr','$state', 'blueprintCreateService','confirmbox', function ($scope,$modal,toastr,$state,bpCreateSer,confirmbox) {
             var blueprintCreation = this;
             //to get the templates listing.
+            $scope.providerType = $state.params.subItem;
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Add the "if" condition .

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 2]
```diff
@@ -76,14 +78,21 @@
                                   <option ng-if="templateSelected.osType" ng-repeat="imageDetail in bpCreate.imageListing" value="{{imageDetail._id}}">{{imageDetail.name}}</option>
                               </select>
                            </div>
+                           <div class="col-lg-6 col-md-6 margintop15" ng-if="providerType=='AZURE'">
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: Replace to "===" from "=="

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 3]
```diff
@@ -93,7 +102,8 @@
                               </label>
                               <select ng-model="bpCreate.newEnt.vpcId" name="vpcId" class="chooseSelect chooseImage width-100 form-control" ng-change="bpCreate.postSubnets();" required>
                                  <option value="">Select VPC</option>
-                                 <option ng-repeat="vpcDetails in bpCreate.vpcListing" value="{{vpcDetails.VpcId}}">{{vpcDetails.VpcId}} ({{vpcDetails.CidrBlock}}) {{vpcDetails.Tags[0].Value}}</option>
+                                 <option ng-if="providerType == 'AWS'" ng-repeat="vpcDetails in bpCreate.vpcListing" value="{{vpcDetails.VpcId}}">{{vpcDetails.VpcId}} ({{vpcDetails.CidrBlock}}) {{vpcDetails.Tags[0].Value}}</option>
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: Replace to "===" from "=="

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 4]
```diff
@@ -103,10 +113,11 @@
                               </label>
                               <select ng-model="bpCreate.newEnt.subnetId" name="subnetId" class="chooseSelect chooseImage width-100 form-control" ng-change="bpCreate.postSecurityGroups();" required>
                                  <option value="">Select Subnet</option>
-                                 <option ng-repeat="subnetDetails in bpCreate.subnetListing" value="{{subnetDetails.SubnetId}}">{{subnetDetails.SubnetId}} ({{subnetDetails.AvailabilityZone}}) | {{subnetDetails.Tags[0].Value}}</option>
+                                 <option ng-if="providerType == 'AWS'" ng-repeat="subnetDetails in bpCreate.subnetListing" value="{{subnetDetails.SubnetId}}">{{subnetDetails.SubnetId}} ({{subnetDetails.AvailabilityZone}}) | {{subnetDetails.Tags[0].Value}}</option>
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: Replace to "===" from "=="

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 5]
```diff
@@ -103,10 +113,11 @@
...
+                                 <option ng-if="providerType == 'AZURE'" ng-repeat="subnetDetailsAzure in bpCreate.subnetListingAzure" value="{{subnetDetailsAzure.Name}}">{{subnetDetailsAzure.Name}}</option>
                               </select>
                            </div>
-                           <div class="col-lg-6 col-md-6 margintop15">
+                           <div class="col-lg-6 col-md-6 margintop15" ng-if="providerType == 'AWS'">
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: Replace to "===" from "=="

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 6]
```diff
@@ -362,17 +440,21 @@
                     if($scope.templateSelected.templatescookbooks){
                         blueprintCreateJSON.runlist = $scope.templateSelected.templatescookbooks.split(',');    
                     }
-
-                    blueprintCreateJSON.securityGroupIds = [];
-                    for(var i =0;i<blueprintCreation.securityGroupListing.length;i++){
-                        if(blueprintCreation.securityGroupListing[i]._isChecked){
-                            blueprintCreateJSON.securityGroupIds.push(blueprintCreation.securityGroupListing[i].GroupId);
-                        }
+                    if($scope.providerType === 'AWS'){
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: convert lowercase to uppercase . ($scope.providerType)

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 7]
```diff
@@ -329,8 +405,11 @@
                     }
 
                     if($scope.bpTypeName === 'OSImage' || $scope.bpTypeName === 'SoftwareStack') {
-                        blueprintCreateJSON.blueprintType = "instance_launch";
-                        blueprintCreateJSON.templateType = $state.params.templateObj.templatetype;
+                        if($scope.providerType === 'AWS'){
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: convert lowercase to uppercase . ($scope.providerType)
 Add a line note

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js (hunk 1 above)
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: convert lowercase to uppercase . ($scope.providerType)
 Add a line note

**Comment by srikanthv02** [comment]
@udeshrl : Changes have been done. Please review it again.

**Pull request merged**